Pixel Game/Assets/Scripts/InventorySlot.cs

```diff
@@ -245,6 +274,54 @@ public void ShowDataOnHover()
 
         // TO-DO - Enable a tooltip box that shows data of the item of THIS current slot
     }
+
+    private void ShowWeaponStats(TextMeshProUGUI itemStatsText)
+    {
+        var weapon = ItemDataGameObject.GetComponent<Weapon>();
+        string powerDiff = "", cooldownDiff = "", projectileSpeedDiff = "";
+        if (gameObject.name != GetEquipmentSlotNameForItem())
+        {
+            powerDiff = GetHoverDiffText(weapon.Damage - playerAttack.CurrentWeapon.Damage);
+            cooldownDiff = GetHoverDiffText(weapon.Cooldown - playerAttack.CurrentWeapon.Cooldown, true);
+            if (weapon.HasProjectileAttack && playerAttack.CurrentWeapon.HasProjectileAttack)
+                projectileSpeedDiff = GetHoverDiffText(weapon.ProjectileAttack.Speed - playerAttack.CurrentWeapon.ProjectileAttack.Speed);
+        }
+        itemStatsText.text = $"Damage: {weapon.Damage} {powerDiff}\n";
+        itemStatsText.text += $"Cooldown: {weapon.Cooldown}s {cooldownDiff}\n";
+        if (weapon.HasProjectileAttack)
+            itemStatsText.text += $"Projectile speed: {weapon.ProjectileAttack.Speed} {projectileSpeedDiff}\n";
+        itemStatsText.text += $"Value: <color=yellow>{ItemDataInSlot.value} coins</color> ";
+    }
+
+    // Returns a string describing the difference between the hovered item
+    // and the currently equipped item
+    // For example, if we have equipped a sword with power 10,
+    // and the weapon we are hovering on has power 15, the text will be "+5" in green
+    private string GetHoverDiffText(float diff, bool reverseColor = false)
+    {
+        string positive = "green";
+        string negative = "#ff7070";
+        string neutral = "white";
+        string color;
+        string sign;
+        if (diff > 0)
+        {
+            color = reverseColor ? negative : positive;
+            sign = "+";
+        }
+        else if (diff == 0)
+        {
+            color = neutral;
+            sign = "+/-";
+        }
+        else
+        {
+            color = reverseColor ? positive : negative;
+            sign = "-";
+        }
+        return $"<color={color}> {sign}{Math.Abs(diff)}</color>";
+    }
+
     public void RemoveDataShowingOnExit()

```

Comment:
Remind me of this function on sunday, I believe we can do this in the inspector. Not sure though

Reply:
So just to be clear, the reason I call it from code is because otherwise it only gets called when the mouse moves. So if you clicked on an item to remove it while holding the mouse still, the hover window would remain.